### ISSUE TITLE
Signal.hammingWindow not implemented correctly

### ISSUE THREAD

**Issue by khoin** [supercollider/supercollider#4323]
The Hamming window is an instance of the Generalized Hamming window with a particular coefficient to cancel out the largest side lobe; it can be seen described here: [J.O.S' SASP](https://ccrma.stanford.edu/~jos/sasp/Hamming_Window.html), or seen implemented elsewhere: [MATLAB](https://www.mathworks.com/help/signal/ref/hamming.html), [Octave](https://hg.savannah.gnu.org/hgweb/octave/file/3cbc3c741f81/scripts/signal/hamming.m#l64), [SciPy](https://github.com/scipy/scipy/blob/v1.0.0/scipy/signal/windows.py#L1036), etc.

I'll pull the equation from Wikipedia:

![Generalized Hamming window family](https://wikimedia.org/api/rest_v1/media/math/render/svg/cff1a2374720284d548f6a59693cd58a297241ec)
where a0 = 0.53836... and a1 = 0.46164... yields the Hamming window.

The current SuperCollider [code](https://github.com/supercollider/supercollider/blob/develop/SCClassLibrary/Common/Math/Signal.sc#L11) for Signal.hammingWindow is this:
```supercollider
^this.newClear(size).fill(0.5).addSine(1, 0.39, -0.5pi);
```

which corresponds to a0 = 0.5 and a1 = 0.39, and is given an example here:
```supercollider
(
Signal.hammingWindow(24).plot("SuperCollider default");
Signal.newClear(24).waveFill({ |x|
	0.5 - (0.39*(2*x).cosPi);
}).plot("Explicit Formula Remade");
)
```

Is this a bug, or is there a rationale behind choosing such coefficient that does not match with the standard definition of Hamming windows? I looked at the [blame](https://github.com/supercollider/supercollider/blame/develop/SCClassLibrary/Common/Math/Signal.sc#L11) and that part hasn't been changed for 17 years. 

In any case, a window should not have its peak not at 1 either.

A proposed fix:
``` supercollider
^this.newClear(size).fill(0.53836).addSine(1, 0.46164, -0.5pi);
```

**Comment by snappizz** [comment]
good catch! there is almost certainly no good reason for this. feel free to file a PR to correct the formula.

**Comment by khoin** [comment]
Since this is fairly old code, I wonder how many projects are dependent
upon it. I don't imagine the change will be drastic, though.

Nonetheless, this to be done tomorrow.

**Comment by telephon** [comment]
I know from people using it. In the past, a good strategy has been: rename the current one `hammingWindow_old` and use `hammingWindow` for the corrected new one.

**Comment by khoin** [comment]
Terribly sorry, but:

I can't seem to propagate my changes in SCClassLibrary to work at all. I've changed the Signal.sc file in `C:\...\SuperCollider-3.10.2\SCClassLibrary\Common\Math\`, press save (Ctrl + S), then Recompile Class Library (Ctrl + Shift + L), then I tried `Signal.hammingWindow(...)` and `Signal.hammingWindow_old` but they either return the old version or `not understood`.

[DEVELOPING.md](https://github.com/supercollider/supercollider/blob/develop/DEVELOPING.md) didn't seem to help. Do I have to go through the build process for this?

**Comment by telephon** [comment]
maybe you haven't made the changes in the library that sclang actually uses. If you open the preferences and open the tab "Interpreter" you may be able to see the path.

**Comment by khoin** [comment]
When I hit Ctrl Shift L, I see the message:
`Compiling directory 'C:\Program Files\SuperCollider-3.10.2\SCClassLibrary'`

which I've double-checked and made sure that the file I'm making changes to is in that directory. Still, it doesn't work. Very strange.

**Comment by khoin** [comment]
Ach. Nevermind. It's because the directory is protected so all changes were rejected, so I needed admin privileges. I don't think SuperCollider IDE reports when it cannot save a file.

**Comment by telephon** [comment]
> I don't think SuperCollider IDE reports when it cannot save a file.
maybe worth another issue!

**Comment by jamshark70** [comment]
File saving fails silently in Linux too. That's awful. #4325

**Comment by khoin** [comment]
This is awaiting the PR. I'll close this.

**Comment by brianlheim** [comment]
Please do not close until PR is merged.

**Comment by khoin** [comment]
Oops. My apologies!

**Comment by brianlheim** [comment]
np! just helps keep track of what we can actually claim as "fixed" :) we have had seemingly innocuous PRs go stale and never merge in the past

**Comment by snappizz** [comment]
fixed in #4324!

**Comment by brianlheim** [comment]
Thanks for that @khoin !
